integ-tests/hadoop/src/main/java/com/google/cloud/bigquery/dwhassessment/sql/SqlHelper.java

```diff
@@ -0,0 +1,114 @@
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** A helper class for reading .sql files. */
+public final class SqlHelper {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(SqlHelper.class);
+  private static final DateTimeFormatter DATE_TIME_FORMATTER =
+      DateTimeFormatter.ofPattern("EEE MMM dd HH:mm:ss zzz yyyy");
+
+  private SqlHelper() {}
+
+  /**
+   * @param sqlPath Path to an .sql file.
+   * @return File contents, never null.
+   */
+  public static String getSql(String sqlPath) {
+    try {
+      return readFileToString(new File(sqlPath), StandardCharsets.UTF_8);
+    } catch (IOException exception) {
+      throw new IllegalStateException(
+          format("Error while reading sql file %s", sqlPath), exception);
+    }
+  }
+
+  /**
+   * @param connection DB connection parameter
+   * @param queries List of strings each of the contains a parametrized SQL request
+   */
+  public static void executeQueries(Connection connection, List<String> queries)
+      throws SQLException {
+    for (String query : queries) {
+      try (Statement stmt = connection.createStatement()) {
+        stmt.execute(query);
+      } catch (SQLException e) {
+        LOGGER.error(String.format("Cannot execute query: %n%s%n", query));
+        throw e;
+      }
+    }
+  }
+
+  /**
+   * @param rs row read from hive
+   * @param schemaBuilder builder of db schema entity
+   */
+  public static void parseSchemaRow(ResultSet rs, Schema.Builder schemaBuilder)
+      throws SQLException {
+    final String DATA_TYPE = "data_type";
+    final String COL_NAME = rs.getString("col_name").trim();
+    switch (COL_NAME) {
+      case "Database:":
+        schemaBuilder.setSchemaName(rs.getString(DATA_TYPE));
+        break;
+      case "Table Type:":
+        schemaBuilder.setType(rs.getString(DATA_TYPE).trim());
+        break;
+      case "CreateTime:":
+        ZonedDateTime zonedDateTimeCT =
+            ZonedDateTime.parse(rs.getString(DATA_TYPE), DATE_TIME_FORMATTER);
+        schemaBuilder.setCreateTime(zonedDateTimeCT.toInstant().getEpochSecond());
+        break;
+      case "LastAccessTime:":
+        if (!rs.getString(DATA_TYPE).trim().equals("UNKNOWN")) {
+          ZonedDateTime zonedDateTimeLAT =
+              ZonedDateTime.parse(rs.getString(DATA_TYPE), DATE_TIME_FORMATTER);
+          schemaBuilder.setLastAccessTime(zonedDateTimeLAT.toInstant().getEpochSecond());
+        }
+        break;
+      case "Owner:":
+        schemaBuilder.setOwner(rs.getString(DATA_TYPE).trim());
+        break;
+      case "Location:":
+        schemaBuilder.setLocation(rs.getString(DATA_TYPE).trim());
+        break;
+      default:
+        if (!COL_NAME.isEmpty() && !COL_NAME.contains(":") && !COL_NAME.contains("#")) {
```

Comment:
Would it be safer to restrict to allowed characters? Should there be a warning if the name is invalid?

Reply:
done